lib/hubspot/properties.rb

```diff
@@ -88,7 +88,7 @@ def filter_results(results, key, include, exclude)
 
       def valid_property_params(params)
         return {} if params.blank?
-        result = params.slice(*PROPERTY_SPECS[:field_names])
+        result = params.with_indifferent_access.slice(*PROPERTY_SPECS[:field_names])
```

Comment:
Is this change required only for the tests to pass? Would it be possible to leave this as is and in our tests pass in a hash `with_indifferent_access`?

Reply:
> Is this change required only for the tests to pass?

It's not required for the tests to pass. I bumped into this when writing tests that pass in a hash with symbol keys instead of string keys. It wasn't obvious to me as to why the request was failing and it took time to realize the keys were being removed because they didn't match the `PROPERTY_SPECS[:field_names]` string keys.

> Would it be possible to leave this as is and in our tests pass in a hash `with_indifferent_access`

Certainly. However, I worry users will encounter the same issue I did (should the user pass in a hash with symbol keys) and spend time digging through this project's source code to figure out why their request is failing. For that reason, I think this change is worth keeping.

Ideally, I think this library doesn't need to be an arbiter of valid field names. That responsibility should fall on the API and any error messages returned by the API will be returned to the library user. 

@cbisnett Do you have any concerns about removing the logic that validates field names?